zebra-state/src/service/finalized_state/disk_format/tests/snapshot.rs

```diff
@@ -0,0 +1,140 @@
+//! Data snapshot tests for RocksDB column families.
+//!
+//! These tests check:
+//! - the name of each column family
+//! - the number of key-value entries
+//! - the bytes in each key and value
+//!
+//! These tests currently use fixed test vectors.
+//!
+//! # Fixing Test Failures
+//!
+//! If this test fails, run `cargo insta review` to update the test snapshots,
+//! then commit the `test_*.snap` files using git.
+//!
+//! # TODO
+//!
+//! Test shielded data, and data activated in Overwinter and later network upgrades.
+
+use std::sync::Arc;
+
+use zebra_chain::{
+    block::Block,
+    parameters::Network::{self, *},
+    serialization::ZcashDeserializeInto,
+};
+
+use crate::{
+    service::finalized_state::{disk_db::DiskDb, disk_format::tests::KV, FinalizedState},
+    Config,
+};
+
+/// Snapshot test for RocksDB column families, and their key-value data.
+///
+/// These snapshots contain the `default` column family, but it is not used by Zebra.
+#[test]
+fn test_raw_rocksdb_column_families() {
+    zebra_test::init();
+
+    test_raw_rocksdb_column_families_with_network(Mainnet);
+    test_raw_rocksdb_column_families_with_network(Testnet);
+}
+
+/// Snapshot raw column families for `network`.
+///
+/// See [`test_raw_rocksdb_column_families`].
+fn test_raw_rocksdb_column_families_with_network(network: Network) {
+    let mut net_suffix = network.to_string();
+    net_suffix.make_ascii_lowercase();
+
+    let mut state = FinalizedState::new(&Config::ephemeral(), Mainnet);
```

Comment:
Shouldn't this use `network` instead of `Mainnet`?

```suggestion
    let mut state = FinalizedState::new(&Config::ephemeral(), network);
```

Reply:
Nice catch!

(We don't validate the network at the state serialization level, that happens in zebra-network and zebra-consensus.)